frontend/src/redux/headlampEventSlice.ts

```diff
@@ -0,0 +1,452 @@
+    resource?: Pod;
+    /** What exactly this event represents. 'OPEN' when the attach dialog is opened. 'CLOSED' when it
+     * is closed.
+     */
+    status: EventStatus.OPENED | EventStatus.CLOSED;
+  };
+}
+
+/**
+ * Event fired when creating a resource.
+ */
+export interface CreateResourceEvent {
+  type: HeadlampEventType.CREATE_RESOURCE;
+  data: {
+    /** What exactly this event represents. 'CONFIRMED' when the user chooses to apply the new resource.
+     * For now only 'CONFIRMED' is sent.
+     */
+    status: EventStatus.CONFIRMED;
+  };
+}
+
+/**
+ * Event fired when there is an error while loading a plugin.
+ */
+export interface PluginLoadingErrorEvent {
+  type: HeadlampEventType.PLUGIN_LOADING_ERROR;
+  data: {
+    /** Information about the plugin. */
+    pluginInfo: {
+      /** The name of the plugin. */
+      name: string;
+      /** The version of the plugin. */
+      version: string;
+    };
+    /** The error that occurred while loading the plugin. */
+    error: Error;
+  };
+}
+
+/**
+ * Event fired when all plugins are loaded.
+ */
+export interface PluginsLoadedEvent {
+  type: HeadlampEventType.PLUGINS_LOADED;
+  data: {
+    /** The list of loaded plugins. */
+    plugins: {
+      /** The name of the plugin. */
+      name: string;
+      /** The version of the plugin. */
+      version: string;
+      /** Whether the plugin is enabled. */
+      isEnabled: boolean;
+    }[];
+  };
+}
+
+/**
+ * Event fired when a resource is loaded in the details view.
+ */
+export interface ResourceDetailsViewLoadedEvent {
+  type: HeadlampEventType.DETAILS_VIEW;
+  data: {
+    /** The resource that was loaded. */
+    resource: KubeObject;
+    /** The error, if an error has occurred */
+    error?: Error;
+  };
+}
+
+/**
+ * Event fired when a list view is loaded for a resource.
+ */
+export interface ResourceListViewLoadedEvent {
+  type: HeadlampEventType.LIST_VIEW;
+  data: {
+    /** The list of resources that were loaded. */
+    resources: KubeObject[];
+    /** The kind of resource that was loaded. */
+    resourceKind: string;
+    /** The error, if an error has occurred */
+    error?: Error;
+  };
+}
+
+/**
+ * Event fired when kubernetes events are loaded (for a resource or not).
+ */
+export interface EventListEvent {
+  type: HeadlampEventType.OBJECT_EVENTS;
+  data: {
+    /** The resource for which the events were loaded. */
+    resource?: KubeObject;
+    /** The list of events that were loaded. */
+    events: Event[];
+  };
+}
+
+export type HeadlampEventCallback = (data: HeadlampEvent) => void;
+
+const initialState: {
+  trackerFuncs: HeadlampEventCallback[];
+} = {
+  trackerFuncs: [],
+};
+
+export const eventAction = createAction<HeadlampEvent>('headlamp/event');
+
+export const listenerMiddleware = createListenerMiddleware();
+listenerMiddleware.startListening({
+  actionCreator: eventAction,
+  effect: async (action, listernerApi) => {
+    // @ts-ignore
```

Comment:
Could you please add a justification as comment to why this is here? (or fix it).
So in the future someone cleaning up can understand why.

Reply:
This was a c&p left-over. I'll fix it.